src/Compilers/CSharp/Test/Semantic/Semantics/StaticNullChecking.cs

```diff
@@ -6578,12 +6579,142 @@ public void Main(string s)
 }
 ", parseOptions: TestOptions.Regular8);
 
+            // PROTOTYPE(NullableReferenceTypes): The suppression operator isn't producing a non-null result
             // PROTOTYPE(NullableReferenceTypes): I'd expect null! to return a non-null result
             VerifyVarLocal(c, "string!");
             c.VerifyTypes();
             c.VerifyDiagnostics();
         }
 
+        [Fact]
+        public void SuppressedObliviousValueGivesNonNullResult()
+        {
+            var libComp = CreateCompilation(@"
+public static class Static
+{
+    public static string Oblivious = null;
+}
+", parseOptions: TestOptions.Regular7);
+
+            var comp = CreateCompilation(@"
+public class C
+{
+    public void Main(string s, string? ns)
+    {
+        s = Static.Oblivious!;
+        var s2 = s;
+        s2/*T:string!*/.ToString(); // ok
+        ns = Static.Oblivious!;
+        ns.ToString(); // ok
+    }
+}
+", references: new[] { libComp.EmitToImageReference() }, parseOptions: TestOptions.Regular8);
+
+            VerifyVarLocal(comp, "string!");
+            comp.VerifyTypes();
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void SuppressedValueGivesNonNullResult()
+        {
+            var comp = CreateCompilation(@"
+public class C
+{
+    public void Main(string? ns, bool b)
+    {
+        var x1 = F(ns!);
+        x1 /*T:string!*/ .ToString();
+
+        var listNS = List.Create(ns);
+        listNS /*T:List<string?>*/ .ToString();
+        var x2 = F2(listNS);
+        x2 /*T:string!*/ .ToString();
+    }
+    public T F<T>(T? x) where T : class => throw null;
+    public T F2<T>(List<T?> x) where T : class => throw null;
+}
+public class List { public static List<T> Create<T>(T t) => throw null; }
+public class List<T> { }
+", parseOptions: TestOptions.Regular8);
+
+            // PROTOTYPE(NullableReferenceTypes): should warn on F2(listNS)
+            comp.VerifyTypes();
```

Comment:
Why would you say this?

Reply:
I think the comment is wrong. I'll remove. Thanks

---
In reply to: [194198102](https://github.com/dotnet/roslyn/pull/27317#discussion_r194198102) [](ancestors = 194198102)